src/Utils/EndpointValidation.ts

```diff
@@ -0,0 +1,40 @@
+export function validateEndpoint(endpointToValidate: string | undefined, allowedEndpoints: string[]): boolean {
+  if (!endpointToValidate) {
+    return true;
+  }
+  const originToValidate: string = new URL(endpointToValidate).origin;
+  const allowedOrigins: string[] = allowedEndpoints.map((allowedEndpoint) => new URL(allowedEndpoint).origin) || [];
```

Comment:
Consider optimizing: if `allowedEndpoint` is static, the origin could be computed once. (as in: `const allowedOrigins = ...`).
This said, it's probably not going to matter much either way.

Reply:
Let's discuss offline. I will leave as is for now.